Let CMake search for the compilers and only fail if they are not found.
See https://gitlab.com/elsi_project/elsi_interface/-/merge_requests/304

Author: Alexander Grund (TU Dresden)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3f016ec7..9aaf8395 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,18 +55,23 @@ SET(INC_PATHS "" CACHE STRING "List of include directories")
 SET(LIBS "" CACHE STRING "List of external libraries to be linked against")
 
 ### Compilers ###
-IF(NOT DEFINED CMAKE_Fortran_COMPILER)
+INCLUDE(CheckLanguage)
+
+CHECK_LANGUAGE(Fortran)
+IF(NOT CMAKE_Fortran_COMPILER)
   MESSAGE(FATAL_ERROR "${MAGENTA}Fortran compiler must be set${COLORRESET}")
 ENDIF()
 ENABLE_LANGUAGE(Fortran)
 
-IF(NOT DEFINED CMAKE_C_COMPILER)
+CHECK_LANGUAGE(C)
+IF(NOT CMAKE_C_COMPILER)
   MESSAGE(FATAL_ERROR "${MAGENTA}C compiler must be set${COLORRESET}")
 ENDIF()
 ENABLE_LANGUAGE(C)
 
 IF(ENABLE_PEXSI)
-  IF(NOT DEFINED CMAKE_CXX_COMPILER)
+  CHECK_LANGUAGE(CXX)
+  IF(NOT CMAKE_CXX_COMPILER)
     MESSAGE(FATAL_ERROR "${MAGENTA}C++ compiler must be set${COLORRESET}")
   ENDIF()
   ENABLE_LANGUAGE(CXX)
